dhis-2/dhis-api/src/main/java/org/hisp/dhis/common/auth/SelfRegistrationParams.java

```diff
@@ -27,24 +27,13 @@
  */
 package org.hisp.dhis.common.auth;
 
-import com.fasterxml.jackson.annotation.JsonProperty;
-import lombok.AllArgsConstructor;
 import lombok.Builder;
-import lombok.Data;
+import lombok.Getter;
 import lombok.NoArgsConstructor;
+import lombok.Setter;
 
-@Data
+@Getter
+@Setter
 @Builder
-@AllArgsConstructor
 @NoArgsConstructor
-public class SelfRegistrationParams {
-  @JsonProperty String username;
-  @JsonProperty String firstName;
-  @JsonProperty String surname;
-  @JsonProperty String password;
-  @JsonProperty String email;
-  @JsonProperty String phoneNumber;
-
-  @JsonProperty("g-recaptcha-response")
-  String recaptchaResponse;
-}
+public class SelfRegistrationParams extends UserRegistrationParams {}
```

Comment:
Can't we reuse the `UserRegistrationParams` if there is nothing more in here?

Reply:
It was implemented this way so that validating a password can accept the abstract class (seeing as they share most of their properties). It could be changed to have 2 validate password methods, One taking the `SelfRegistrationParams` and the other taking `CompleteRegistrationParams`.